packages/pluggableWidgets/image-viewer-web/src/ImageViewer.tsx

```diff
@@ -0,0 +1,71 @@
+import { ValueStatus } from "mendix";
+import { createElement, FunctionComponent } from "react";
+import { ImageViewerContainerProps } from "../typings/ImageViewerProps";
+import { ImageViewer as ImageViewerComponent } from "./components/ImageViewer";
+
+type ImageProps = {
+    type: "image" | "icon";
+    image: string | undefined;
+};
+
+function getImageProps({ datasource, imageIcon, imageObject, imageUrl }: ImageViewerContainerProps): ImageProps {
+    const fallback: ImageProps = {
+        type: "image",
+        image: undefined
+    };
+    switch (datasource) {
+        case "image":
+            return {
+                type: "image",
+                image: imageObject?.status === ValueStatus.Available ? imageObject.value.uri : undefined
```

Comment:
This implementation can cause flickering when a new image is loaded again. During load time the previous image can be displayed.

Reply:
I will discuss this with Chris whether it's relevant for this initial version. Because there are a lot of ways we can handle this, but I am not sure how "locked in" we want to make this.

Reply:
Papertrail: discussed during standup that we will omit this from initial version as this requires some more UX thinking